src/parser.rs

```diff
@@ -345,13 +345,19 @@ impl<'tokens, 'src> Parser<'tokens, 'src> {
               items.push(Item::Assignment(self.parse_assignment(false)?));
             } else {
               let doc = pop_doc_comment(&mut items, eol_since_last_comment);
-              items.push(Item::Recipe(self.parse_recipe(doc, false)?));
+              items.push(Item::Recipe(self.parse_recipe(doc, false, false)?));
             }
           }
         }
       } else if self.accepted(At)? {
         let doc = pop_doc_comment(&mut items, eol_since_last_comment);
-        items.push(Item::Recipe(self.parse_recipe(doc, true)?));
+        items.push(Item::Recipe(self.parse_recipe(doc, true, false)?));
+      } else if self.accepted(BracketL)? {
+        self.parse_attribute()?;
+        self.expect_eol()?;
+        let quiet = if self.accepted(At)? { true } else { false };
+        let doc = pop_doc_comment(&mut items, eol_since_last_comment);
+        items.push(Item::Recipe(self.parse_recipe(doc, quiet, true)?));
```

Comment:
I think it's a little confusing that `parse_attribute` consumes the `BracketR` but not the `BracketL`. How about something like this:

(Also, `if CONDITION { true } else { false }` is equivalent to `CONDITION`.)

```suggestion
      } else if self.accepted(BracketL)? {
        let attribute_name = self.parse_attribute_name()?
        self.expect(BracketR)?;
        self.expect_eol()?;
        let quiet = self.accepted(At)?;
        let doc = pop_doc_comment(&mut items, eol_since_last_comment);
        items.push(Item::Recipe(self.parse_recipe(doc, quiet, true)?));
```